hdfs/__main__.py

```diff
@@ -36,12 +36,14 @@
                                 0 allocates a thread per file. [default: 0]
   -v --verbose                  Enable log output. Can be specified up to three
                                 times (increasing verbosity each time).
+  -r REPLICATION --replication=REPLICATION  Set the expected HDFS replication count [default: 3]
```

Comment:
I missed your reply and switched to other stuff. :-(

> The current code doesn't set the default.

I'm super confused with that docs-as-a-config thing, but as I understand, `[default: 3]` sets the defaults?

> Can you keep this behavior when the option isn't set? The cluster might override the default site-wide.

Not sure I follow. The cluster does have site-wide settings but the client can override them. So if you don't pass that fro the client perspective - cluster defaults will be used.

Reply:
The current code (before this PR) doesn't set the default on the client, so the cluster's default will be used. This is a good behavior which would be lost after this change since the client now sets a default of 3. Instead the client should only set replication when explicitly specified by the user.